x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/trained_models_deployment_modal.tsx

```diff
@@ -8,179 +8,179 @@ import { EuiConfirmModal, useGeneratedHtmlId, EuiHealth } from '@elastic/eui';
 import React from 'react';
 
 import { EuiLink } from '@elastic/eui';
-import { useEffect, useState } from 'react';
-import type { SharePluginStart } from '@kbn/share-plugin/public';
+import { useEffect, useMemo, useState } from 'react';
 import { i18n } from '@kbn/i18n';
+import { deNormalize } from '../../../../components/mappings_editor/lib';
+import { Field, SemanticTextField } from '../../../../components/mappings_editor/types';
+import { useMLModelNotificationToasts } from '../../../../../hooks/use_ml_model_status_toasts';
+import { useMappingsState } from '../../../../components/mappings_editor/mappings_state_context';
+import { useAppContext } from '../../../../app_context';
 
-interface SemanticTextProps {
-  setIsModalVisible: (isVisible: boolean) => void;
-  refreshModal: () => void;
-  pendingDeployments: Array<string | undefined>;
-  errorsInTrainedModelDeployment: string[];
-  url?: SharePluginStart['url'];
+export interface TrainedModelsDeploymentModalProps {
+  fetchData: () => void;
+  errorsInTrainedModelDeployment: Record<string, string | undefined>;
+  setErrorsInTrainedModelDeployment: React.Dispatch<
+    React.SetStateAction<Record<string, string | undefined>>
+  >;
 }
 
 const ML_APP_LOCATOR = 'ML_APP_LOCATOR';
 const TRAINED_MODELS_MANAGE = 'trained_models';
 
+function isSemanticTextField(field: Partial<Field>): field is SemanticTextField {
```

Comment:
This function seems to be declared and used in multiple places - can we move it to [utils.ts ](https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_management/public/application/components/mappings_editor/lib/utils.ts)?

Reply:
fixed in https://github.com/elastic/kibana/commit/85da386e1e774d01daf5e58fee2e5f19236ab52b